ui/src/components/directory/ParentHeader.vue

```diff
@@ -98,14 +99,13 @@ function getColour(entity: any) {
   return "color: " + getColourFromType(entity[RDF.TYPE]);
 }
 
-async function isSelectableEntity(): Promise<boolean> {
-  if (props.validationQuery) {
-    const queryRequest = _.cloneDeep(props.validationQuery);
-    queryRequest.textSearch = props.entity[RDFS.LABEL];
-    const queryResults = await QueryService.queryIM(queryRequest);
-    if (queryResults.entities && queryResults.entities.findIndex(item => item.iri === props.entity[RDFS.LABEL])) return true;
-    else return false;
-  } else return true;
+async function getIsSelectableEntity(): Promise<boolean> {
+  if (!props.validationQuery) return true;
+  const queryRequest = _.cloneDeep(props.validationQuery);
+  queryRequest.textSearch = props.entity[RDFS.LABEL];
+  const queryResults = await QueryService.queryIM(queryRequest);
+  if (!isArrayHasLength(queryResults.entities)) return false;
```

Comment:
looks like it could fail if queryResults is undefined. 2 checks required, is queryRequest exists and if entities is array